blocks/library/embed/index.js

```diff
@@ -19,17 +19,21 @@ registerBlock( 'core/embed', {
 		caption: children( 'figcaption' )
 	},
 
-	edit( { attributes, isSelected, setAttributes } ) {
+	edit( { attributes, setAttributes, focus, setFocus } ) {
 		const { url, title, caption } = attributes;
 
 		return (
 			<figure>
-				<iframe src={ url } title={ title } />
-				{ caption || isSelected ? (
+				<div className="iframe-overlay">
+					<iframe src={ url } title={ title } />
```

Comment:
Are we talking about having many different block types for various sorts of embeds? I'm not sure why this would be a common behavior across blocks, at least to the point of wanting to create an abstraction.